### PR TITLE
[Naming] Skip DateTimeImmutable on RenamePropertyToMatchTypeRector

### LINKED ISSUE #8344: Incorrect behavior of RenamePropertyToMatchTypeRector
# Bug Report

| Subject        | Details              |
| :------------- | :--------------------|
| Rector version | last dev-main      |
| Installed as   | composer dependency  |

## Minimal PHP Code Causing Issue

See https://getrector.com/demo/49bb6ed0-cb01-484d-8fab-c9917d7e1232

```php
<?php

final class DemoFile
{   
    public \DateTimeImmutable $created;
    
    public function __construct(
        public \DateTimeImmutable $changed,
    ) {}
}

```

### Responsible rules

* `RenamePropertyToMatchTypeRector`

## Expected Behavior

I like that the `NAMING` setlist makes my variable names more descriptive. With that in mind, I would not expect the name change from `$changed` to `$dateTimeImmutable`.

I had two more findings regarding these rules:
- Property names of doctrine entities are changed as well, it would be great if these can be skipped to prevent the need for a database migration.
- I cannot reproduce yet on the playground, but In my code I have multiple `\DateTimeImmutable` properties names in the same class changed to `$dateTimeImmutable`, which leads to duplicate property names:

(partial diff)
```
1 file with changes
===================

1) src/Domain/Organisation/Organisation.php:65

    ---------- begin diff ----------
@@ @@
     #[ORM\Column(type: Types::DATETIMETZ_IMMUTABLE)]
-    private \DateTimeImmutable $created;
+    private \DateTimeImmutable $dateTimeImmutable;
@@ @@
-        private ?\DateTimeImmutable $activeUntil,
+        private ?\DateTimeImmutable $dateTimeImmutable,
@@ @@
-        $this->created = $clock->now();
+        $this->dateTimeImmutable = $clock->now();
    ----------- end diff -----------

Applied rules:
 * RenameParamToMatchTypeRector
 * RenameVariableToMatchNewTypeRector
 * RenamePropertyToMatchTypeRector
```

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector/issues/8344

### CODE REVIEW

**Comment by samsonasik** [comment]
Fixed 🎉 /cc @darthf1

File: rules/Naming/ExpectedNameResolver/MatchParamTypeExpectedNameResolver.php [hunk 1]
```diff
@@ -28,9 +30,9 @@ public function resolve(Param $param): ?string
 
         $staticType = $this->staticTypeMapper->mapPhpParserNodePHPStanType($param->type);
 
-        // include nullable too
+        // include nullabgitle too
         // skip date time + date time interface, as should be kept
-        if ($this->isDateTimeType($staticType)) {
+        if ($this->nodeTypeResolver->isObjectType($param->type, new ObjectType('DateTimeInterface'))) {
```
Comment (samsonasik) [inline rules/Naming/ExpectedNameResolver/MatchParamTypeExpectedNameResolver.php:35]: @TomasVotruba using `$this->nodeTypeResolver->isObjectType()` handle it correctly as `DateTimeImmutable` is child of `DateTimeInterface` 👍

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I am merging it ;)

**Comment by darthf1** [comment]
Thanks @samsonasik!

**Pull request merged**